File: core/src/ru/myitschool/dcrawler/math/FunctionAction.java
package ru.myitschool.dcrawler.math;

/**
 * Created by Voyager on 09.05.2018.
 */
public class FunctionAction extends MathAction {

    Function function;
    String descripton;

    public FunctionAction(Function function, String descripton) {
        this.function = function;
        this.descripton = descripton;
    }

    @Override
    public int act() {
        return 0;
    }

    @Override
    public String getDescription() {
        return null;
    }

    public interface Function{
        int getNumber(Object[] args);
    }
}

File: core/src/com/wolg_vlad/dcrawler/math/FunctionAction.kt
package com.wolg_vlad.dcrawler.math

/**
 * Created by Voyager on 09.05.2018.
 */
class FunctionAction(var function: Function, var descripton: String) : MathAction() {
    override fun act(): Int {
        return 0
    }

    override val description: String?
        get() = null

    interface Function {
        fun getNumber(args: Array<Any?>?): Int
    }

}